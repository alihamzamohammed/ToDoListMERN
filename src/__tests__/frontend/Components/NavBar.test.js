import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import { BrowserRouter as Router } from "react-router-dom";
import NavBar from "../../../Components/Navbar";

describe("Navbar Tests", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("Renders the navbar", () => {
    act(() => {
      render(
        <Router>
          <NavBar></NavBar>
        </Router>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
});
