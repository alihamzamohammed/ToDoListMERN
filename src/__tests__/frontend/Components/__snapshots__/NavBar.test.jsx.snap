// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Tests Renders the navbar 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <span
      class="navbar-brand"
    >
      Todo
    </span>
    <button
      aria-controls="navbar"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="navbar"
    >
      <div
        class="mr-auto navbar-nav"
      >
        <a
          class="active nav-link"
          data-rb-event-key="/"
          href="/"
          id="nav-home"
        >
          Home
        </a>
        <div
          class="nav-item dropdown"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle nav-link"
            href="#"
            id="create-dropdown"
            role="button"
          >
            Create
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
